### DISCUSSION TITLE
List object as root

### DISCUSSION THREAD

**Discussion by taichi-ishitani** [apple/pkl#182] [Ideas]
For YAML and JSON, a list object can be the root object lie below.
![image](https://github.com/apple/pkl/assets/2922232/aa20638a-94fa-4558-bbed-87ddb885a288)

How about adding such structure support. I think current Plk does not support it.

**Comment by johnallen3d** [comment]
You could use [Module Output](https://pkl-lang.org/main/current/language-reference/index.html#module-output) for this:

```pkl
// foo.pkl
list {
  "foo"
  "bar"
}

output {
  value = list
}
```

```bash
❯ pkl eval foo.pkl --format yaml
- foo
- bar
```

**Reply by taichi-ishitani** [reply]
Thank you for your comment @johnallen3d .
But I'd like to describe this kind of structure directly like below.

```
0
1
2
```
